### PR TITLE
Add tolerations support

### PR DESCRIPTION
**Pull request by worr**
| Q               | A
| --------------- | ---
| Bug fix?        | no|yes
| New feature?    | yes|yes
| API breaks?     | no|yes
| Deprecations?   | no|yes
| Related tickets | none
| License         | Apache 2.0

### What's in this PR?
Adds support for tolerations in the logging operator. Now, in both in the helm charts and the CRDs, you can specify tolerations for `fluentbit` and `fluentd`

### Why?
While this doesn't fix an outstanding issue, this does make it possible to schedule `fluentbit` on nodes that may have been tainted as perhaps administrative nodes.

### Additional context
Testing:

* built operator and pushed custom docker image
* deployed operator on a kubernetes cluster with tainted nodes
* deployed fluentbit and fluentd CRDs via helm with toleration arguments provided

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [ ] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [ ] Logging code meets the guideline (TODO)
- [ ] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

File: charts/logging-operator-fluent/templates/fluentbit-cr.yaml [hunk 1]
```diff
@@ -16,8 +16,9 @@ spec:
     prometheus.io/port:   "2020"
   image: {{ toYaml .Values.fluentbit.image | nindent 4 }}
   resources: {}
+  tolerations: {{ toYaml .Values.fluentbit.tolerations | nindent 4 }}
```
Comment (pepov) [inline charts/logging-operator-fluent/templates/fluentbit-cr.yaml]: I would ask to put this in an if statement to leave the tolerations off if not defined

File: charts/logging-operator-fluent/templates/fluentd-cr.yaml [hunk 2]
```diff
@@ -19,9 +19,10 @@ spec:
   configReloaderImage: {{ toYaml .Values.fluentd.configReloaderImage | nindent 4 }}
   resources: {}
   fluentdPvcSpec: {{ toYaml .Values.fluentd.fluentdPvcSpec | nindent 4 }}
+  tolerations: {{ toYaml .Values.fluentd.tolerations | nindent 4 }}
```
Comment (pepov) [inline charts/logging-operator-fluent/templates/fluentd-cr.yaml]: I would ask to put this in an if statement to leave the tolerations off if not defined

File: pkg/apis/logging/v1alpha1/fluentbit_types.go [hunk 3]
```diff
@@ -37,6 +37,7 @@ type FluentbitSpec struct {
 	Image       ImageSpec                   `json:"image"`
 	TLS         FluentbitTLS                `json:"tls"`
 	Resources   corev1.ResourceRequirements `json:"resources,omitempty"`
+	Tolerations []corev1.Toleration         `json:"tolerations"`
```
Comment (pepov) [inline pkg/apis/logging/v1alpha1/fluentbit_types.go]: I would ask to use omitempty here.

File: pkg/apis/logging/v1alpha1/fluentd_types.go [hunk 4]
```diff
@@ -41,6 +41,7 @@ type FluentdSpec struct {
 	ConfigReloaderImage ImageSpec                        `json:"configReloaderImage"`
 	Resources           corev1.ResourceRequirements      `json:"resources,omitempty"`
 	ServiceType         corev1.ServiceType               `json:"serviceType,omitempty"`
+	Tolerations         []corev1.Toleration              `json:"tolerations"`
```
Comment (pepov) [inline pkg/apis/logging/v1alpha1/fluentd_types.go]: I would ask to use omitempty here.

**Comment by worr** [comment]
I've pushed the requested fixes

**Comment by pepov** [comment]
Thanks!

**Pull request merged**